curvlinops/kfac.py

```diff
@@ -757,3 +769,161 @@ def compute_parameter_mapping(
+
+        Will call ``_compute_kfac`` if it has not been called before and will cache the
+        determinant until ``_compute_kfac`` is called again. Uses the property of the
+        Kronecker product that :math:`\det(A \otimes B) = \det(A)^{m} \det(B)^{n}`,
+        where
+        :math:`A \in \mathbb{R}^{n \times n}` and :math:`B \in \mathbb{R}^{m \times m}`.
+
+        Returns:
+            Determinant of the KFAC approximation.
+        """
+        if self._det is None:
+
+            if not self._input_covariances and not self._gradient_covariances:
+                self._compute_kfac()
+
+            self._det = 1.0
+            for mod_name, param_pos in self._mapping.items():
+                m = len(self._gradient_covariances[mod_name])
+                det_ggT = self._gradient_covariances[mod_name].det()
+                if (
+                    not self._separate_weight_and_bias
+                    and "weight" in param_pos.keys()
+                    and "bias" in param_pos.keys()
+                ):
+                    n = len(self._input_covariances[mod_name])
+                    det_aaT = self._input_covariances[mod_name].det()
+                    self._det *= det_aaT.pow(m) * det_ggT.pow(n)
+                else:
+                    for p_name in param_pos.keys():
+                        if p_name == "weight":
+                            n = len(self._input_covariances[mod_name])
+                            self._det *= self._input_covariances[mod_name].det().pow(m)
+                        else:
+                            n = 1
+                        self._det *= det_ggT.pow(n)
+
+        return self._det
+
+    @property
+    def logdet(self) -> Tensor:
+        r"""Log determinant of the KFAC approximation.
+
+        Will call ``_compute_kfac`` if it has not been called before and will cache the
+        log determinant until ``_compute_kfac`` is called again. Uses the property of
+        the Kronecker product that
+        :math:`\log \det(A \otimes B) = m \log \det(A) + n \log \det(B)`, where
+        :math:`A \in \mathbb{R}^{n \times n}` and :math:`B \in \mathbb{R}^{m \times m}`.
+        More numerically stable than the ``det`` property.
+
+        Returns:
+            Log determinant of the KFAC approximation.
+        """
+        if self._logdet is None:
+
+            if not self._input_covariances and not self._gradient_covariances:
+                self._compute_kfac()
+
+            self._logdet = 0.0
+            for mod_name, param_pos in self._mapping.items():
+                m = len(self._gradient_covariances[mod_name])
+                logdet_ggT = self._gradient_covariances[mod_name].logdet()
+                if (
+                    not self._separate_weight_and_bias
+                    and "weight" in param_pos.keys()
+                    and "bias" in param_pos.keys()
+                ):
+                    n = len(self._input_covariances[mod_name])
```

Comment:
I recently decided to use `len` instead of `shape[0]` because it is *slightly* [faster for numpy](https://stackoverflow.com/questions/51332056/numpy-performance-gap-between-lenarr-and-arr-shape0) and since everything is typed anyway, it is usually clear what types objects have. But I just checked and for torch `shape[0]` seems to be slightly faster, so I switched to this -- and I agree that it is making it even more explicit that it is a tensor object.
Not that the performance of this matters at all, but had to somehow choose what to use haha